src/request-api.js

```diff
@@ -106,51 +112,60 @@ function requestAPI (config, path, args, qs, files, buffer, cb) {
     opts.headers['User-Agent'] = config['user-agent']
   }
 
-  if (files) {
+  if (options.files) {
     if (!stream.boundary) {
-      return cb(new Error('No boundary in multipart stream'))
+      return callback(new Error('No boundary in multipart stream'))
     }
 
     opts.headers['Content-Type'] = `multipart/form-data; boundary=${stream.boundary}`
-    opts.downstreamRes = stream
     opts.payload = stream
   }
 
-  return Wreck.request(opts.method, opts.uri, opts, onRes(buffer, cb, opts.uri))
+  return Wreck.request(opts.method, opts.uri, opts, onRes(options.buffer, callback, opts.uri))
 }
 
-// -- Interface
+//
+// -- Module Interface
 
 exports = module.exports = function getRequestAPI (config) {
-  var send = function (path, args, qs, files, buffer, cb) {
-    if (typeof buffer === 'function') {
-      cb = buffer
-      buffer = false
+  /*
+   * options: {
+   *   path:   // API path (like /add or /config)
+   *   args:   // Arguments to the command
+   *   qs:     // Opts as query string opts to the command --something
+   *   files:  // files to be sent
+   *   buffer: // buffer the request before sending it
+   * }
+   */
+  const send = function (options, callback) {
+    if (typeof options !== 'object') {
+      return callback(new Error('no options were passed'))
     }
 
-    if (typeof cb !== 'function' && typeof Promise !== 'undefined') {
+    if (typeof callback !== 'function' && typeof Promise !== 'undefined') {
       return new Promise(function (resolve, reject) {
```

Comment:
Maybe we can drop this and move to `promisify` on all methods or something like that?